src/Workspaces/CSharp/Portable/EmbeddedLanguages/VirtualChars/CSharpVirtualCharService.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.EmbeddedLanguages.VirtualChars
+{
+    [ExportLanguageService(typeof(IVirtualCharService), LanguageNames.CSharp), Shared]
+    internal class CSharpVirtualCharService : AbstractVirtualCharService
+    {
+        public static readonly IVirtualCharService Instance = new CSharpVirtualCharService();
+
+        protected override bool IsStringLiteralToken(SyntaxToken token)
+            => token.Kind() == SyntaxKind.StringLiteralToken;
+
+        protected override ImmutableArray<VirtualChar> TryConvertToVirtualCharsWorker(SyntaxToken token)
+        {
+            Debug.Assert(!token.ContainsDiagnostics);
+            if (token.Kind() == SyntaxKind.StringLiteralToken)
+            {
+                return token.IsVerbatimStringLiteral()
+                    ? TryConvertVerbatimStringToVirtualChars(token, "@\"", "\"", escapeBraces: false)
+                    : TryConvertStringToVirtualChars(token, "\"", "\"", escapeBraces: false);
+            }
+            else if (token.Kind() == SyntaxKind.InterpolatedStringTextToken)
```

Comment:
@jcouv This code is the same as when you originally reviewed *except* that i've added support for interpolated strings.  this is so that we can effectively classify escape sequences in something like  ```$"{foo}\r\n{bar}"```.  Note: interpolations are actually *simpler* to handle because the interpolation text is *only* the ```\r\n``` art, and doesn't include things like ```$"``` that you then have to ignore.